interactive/views.py

```diff
@@ -48,20 +48,29 @@ def register_interest_done(request):
 
 @login_required
 def new_analysis_request(request):
-    codelists = [("", "---")] + opencodelists.fetch()
+    codelists = opencodelists.fetch()
+
+    def codelists_to_choices(codelists):
+        yield ("", "---")
+        for codelist in codelists:
+            yield (codelist["slug"], codelist["name"])
+
+    codelist_choices = list(codelists_to_choices(codelists))
+
     if request.method == "POST":
-        form = AnalysisRequestForm(request.POST, codelists=codelists)
+        form = AnalysisRequestForm(request.POST, codelists=codelist_choices)
         if form.is_valid():
             analysis_request = form.save(user=request.user)
             submit_analysis(analysis_request)
             return redirect("request_analysis_done")
     else:
-        form = AnalysisRequestForm(codelists=codelists)
+        form = AnalysisRequestForm(codelists=codelist_choices)
 
     ctx = {
         "form": form,
         "start_date": START_DATE,
         "end_date": END_DATE,
+        "codelists": codelists,
```

Comment:
Django question: why does codelist need to be added to the context, if it's also on the form?

Reply:
This is the "full" list of dicts that `codelists.fetch()` returns, it's used to pass that data to the React component.  The form has a `codelists` field which is populated with the choices we construct on line 58.